src/main/java/uk/gov/hmcts/reform/sendletter/util/CsvWriter.java

```diff
@@ -108,6 +117,41 @@ public static File writeStaleLettersReport(Stream<BasicLetterInfo> letters) thro
         return csvFile;
     }
 
+    public static Map<String, File> writeFtpLettersToCsvFiles(
+        Map<String, List<FileInfo>> serviceToLetters
+    ) throws IOException {
+        Map<String, File> csvFilesForServices = new HashMap<>();
+
+        for (Map.Entry<String, List<FileInfo>> entry : serviceToLetters.entrySet()) {
+            var path = Files.createTempFile(
+                format("FTP-uploaded-letters-%s", entry.getKey()), ".csv", ATTRIBUTE);// Compliant
+
+            var csvFile = path.toFile();
+            CSVFormat csvFileHeader
+                = CSVFormat.DEFAULT.builder().setHeader(FTP_UPLOADED_LETTERS_EMAIL_CSV_HEADERS).build();
+            var count = new AtomicInteger(0);
```

Comment:
Looks like this count should be defined before the 'for...' loop, otherwise it is initialised in each iteration.